Sentinel/Classes/Core/PreferencesInfo/PreferenceItem+Init.swift

```diff
@@ -0,0 +1,169 @@
+//
+//  PreferenceItem+Init.swift
+//  Sentinel
+//
+//  Created by Infinum on 12.09.2023..
+//
+
+import UIKit
+
+extension PreferenceItem {
+
+    /// Preference as a String. Displays as UITextField.
+    ///
+    /// - Parameter name: The name of the preference
+    /// - Parameter info: Optional additional info like instructions, format etc.
+    /// - Parameter userDefaultsKey: The key under which this preference is saved in UserDefaults
+    /// - Parameter userDefaults: Optionally define a different suite of UserDefaults (standard by default).
+    /// - Parameter onDidSet: Optional execution block once the value is changed.
+    public convenience init(
```

Comment:
Based on your changes, I suggest making this more generic.
I recommend injecting a type (enum) instead of having multiple inits (for text, bool, etc.). 

Also, I would go with a `Validator` protocol and inject different validators for different types (it can be an array of validators if you like). E.g., what happens now if the text should be all uppercase and the length should be max 5 characters? It is hard to maintain current implementation by adding additional parameters to the init method.

It would be good to add getter and setter in the init (like bool).

Reply:
In short, you'll have basically one init instead of multiple ones.

Reply:
This was the original idea. The problem was however they type of the value in the `onDidSet` closure - it was `String?` and not `Double` or `Int`. Having these convenience initializers enabled me to define the exact type that is handled.

Otherwise it would have to be a generic initializer, and I would have to check whether T is supported (String, Int, Double...)
```swift
convenience init<T>(name: String, inputType: InputType, type: T.Type, onDidSet: (T?) -> Void) 
```